third_party/2and3/decorator.pyi

```diff
@@ -0,0 +1,90 @@
+# Stubs for decorator (Python 2 and 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+import sys
+from typing import Any, Callable, Dict, List, NamedTuple, Optional, Pattern, Text, Tuple, TypeVar
+
+_C = TypeVar("_C", bound=Callable[..., Any])
+_Func = TypeVar("_Func", bound=Callable[..., Any])
+_Caller = TypeVar("_Caller", bound=Callable[..., Any])
+
+def get_init(cls): ...
+
+# Duplicated from `stdlib/3/inspect.pyi`
+FullArgSpec = NamedTuple('FullArgSpec', [('args', List[str]),
+                                         ('varargs', Optional[str]),
+                                         ('varkw', Optional[str]),
+                                         ('defaults', tuple),
+                                         ('kwonlyargs', List[str]),
+                                         ('kwonlydefaults', Dict[str, Any]),
+                                         ('annotations', Dict[str, Any]),
+                                         ])
```

Comment:
This is only available in Python 2 in the implementation, so let's move it into the conditional below.

Reply:
argh I'm not familiar with PRs, my other comment got lost.
If I move it into the conditional, and import `getfullargspec` from `inspect`, I get the following mypy error

```
/usr/local/Cellar/python@2/2.7.14_3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/decorator.py:47: error: Revealed type is 'def (func: builtins.object) -> Tuple[builtins.list[builtins.str], Union[builtins.str, None], Union[builtins.str, None], builtins.tuple[Any], builtins.list[builtins.str], builtins.dict[builtins.str, Any], builtins.dict[builtins.str, Any], fallback=inspect.FullArgSpec]'
/usr/local/Cellar/python@2/2.7.14_3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/decorator.py:56: error: All conditional function variants must have identical signatures
```

the signatures were almost idential, except for `inspect.FullArgSpec` vs `decorator.FullArgSpec`.
Duplicating `FullArgSpec` here is hacky; it allows us to define `getfullargspec` under >=py3 as having `decorator.FullArgSpec`.

Do you have a suggestion for a fix? Maybe a well-placed `cast` could do what I need...

Reply:
I think if you move it into the Python 2 block and follow my remarks below, it should work now.